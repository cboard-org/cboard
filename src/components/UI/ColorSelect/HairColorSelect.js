import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape } from 'react-intl';
import FormControl from '@material-ui/core/FormControl';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import CloseIcon from '@material-ui/icons/Close';

import IconButton from '../IconButton';
import Circle from './Circle';
import messages from './HairColor.messages';

const hairColorSources = new Map([
  [
    'arasaac',
    [
      {
        name: 'blonde',
        color: '#FDD700'
      },
      {
        name: 'brown',
        color: '#A65E26'
      },
      {
        name: 'darkBrown',
        color: '#6A2703'
      },
      {
        name: 'gray',
        color: '#EFEFEF'
      },
      {
        name: 'darkGray',
        color: '#AAABAB'
      },
      {
        name: 'red',
        color: '#ED4120'
      },
      {
        name: 'black',
        color: '#020100'
      }
    ]
  ]
]);

const propTypes = {
  source: PropTypes.string.isRequired,
  intl: intlShape.isRequired,
  onChange: PropTypes.func.isRequired,
  selectedColor: PropTypes.string.isRequired
};

class HairColorSelect extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      hairColorMenu: hairColorSources.has(props.source)
        ? hairColorSources.get(props.source)
        : hairColorSources.get('arasaac')
    };
  }

  render() {
    const { intl, onChange, selectedColor } = this.props;
    const hairColorLabel = intl.formatMessage(messages.hairColor);
    const radioGroupStyle = { flexDirection: 'row' };
    const radioItemStyle = { padding: '2px' };

    return (
      <FormControl className="ColorSelect">
        <RadioGroup
          aria-label={hairColorLabel}
          name="hairColor"
          value={selectedColor}
          style={radioGroupStyle}
          onChange={onChange}
        >
          {this.state.hairColorMenu.map(hairColor => (
            <Radio
              key={hairColor.name}
              value={hairColor.color}
              style={radioItemStyle}
              icon={<Circle color={hairColor.color} />}
              checkedIcon={<Circle color={hairColor.color} />}
            />
          ))}
          {selectedColor && (
            <IconButton
              label={intl.formatMessage(messages.clearSelection)}
              onClick={() => {
                onChange();
              }}
            >
              <CloseIcon />
            </IconButton>
          )}
        </RadioGroup>
      </FormControl>
    );
  }
}

HairColorSelect.propTypes = propTypes;
export default injectIntl(HairColorSelect);
