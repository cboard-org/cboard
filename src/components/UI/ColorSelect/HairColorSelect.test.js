import React from 'react';
import { mount } from 'enzyme';
import { shallowMatchSnapshot } from '../../../common/test_utils';
import HairColorSelect from './HairColorSelect';

jest.mock('./HairColor.messages', () => {
  return {
    hairColor: {
      id: 'cboard.components.HairColorSelect.hairColor',
      defaultMessage: 'Skin Tone'
    },
    clearSelection: {
      id: 'cboard.components.HairColorSelect.clearSelection',
      defaultMessage: 'Clear selection'
    }
  };
});

const intl = { formatMessage: () => 'dummy translation' };
const HAIR_COLORS = [
  '#FDD700',
  '#A65E26',
  '#6A2703',
  '#EFEFEF',
  '#AAABAB',
  '#ED4120',
  '#020100'
];

describe('HairColorSelect tests', () => {
  const props = {
    source: 'arasaac',
    intl,
    onChange: () => {},
    selectedColor: HAIR_COLORS[0]
  };

  test('default renderer', () => {
    shallowMatchSnapshot(<HairColorSelect {...props} />);
  });
  test('mount renderer', () => {
    mount(<HairColorSelect {...props} />);
  });
});
