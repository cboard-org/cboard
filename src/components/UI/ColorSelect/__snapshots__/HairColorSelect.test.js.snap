// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HairColorSelect tests default renderer 1`] = `
<HairColorSelect
  intl={
    Object {
      "formatMessage": [Function],
    }
  }
  onChange={[Function]}
  selectedColor="#FDD700"
  source="arasaac"
/>
`;
